src/general/Mutations.cs

```diff
@@ -170,96 +170,132 @@ private void MutateBehaviour(Species parent, Species mutated)
-                            AddNewOrganelle(mutatedOrganelles,
-                                organellesThatCanBeDuplicated.Random(random).Definition);
-                        }
-                        else
-                        {
-                            AddNewOrganelle(mutatedOrganelles, GetRandomOrganelle(isBacteria));
-                        }
+                        AddNewOrganelle(mutatedOrganelles, GetRandomOrganelle(isBacteria));
                     }
                 }
             }
+        }
 
-            if (isBacteria)
+        if (isBacteria)
+        {
+            if (random.Next(0.0f, 1.0f) <= Constants.MUTATION_BACTERIA_TO_EUKARYOTE)
             {
-                if (random.Next(0.0f, 1.0f) <= Constants.MUTATION_BACTERIA_TO_EUKARYOTE)
-                {
-                    AddNewOrganelle(mutatedOrganelles, nucleus);
-                }
+                AddNewOrganelle(mutatedOrganelles, nucleus);
             }
+        }
+
+        // Disallow creating empty species as that throws an exception when trying to spawn
+        if (mutatedOrganelles.Count < 1)
+        {
+            // Add the first parent species organelle
+            AddNewOrganelle(mutatedOrganelles, parentOrganelles[0].Definition);
 
-            // Disallow creating empty species as that throws an exception when trying to spawn
+            // If still empty, copy the first organelle of the parent
             if (mutatedOrganelles.Count < 1)
+                mutatedOrganelles.Add((OrganelleTemplate)parentOrganelles[0].Clone());
+        }
+
+        var islandHexes = mutatedOrganelles.GetIslandHexes();
+
+        // Attach islands
+        while (islandHexes.Count > 0)
+        {
+            var mainHexes = mutatedOrganelles.ComputeHexCache().Except(islandHexes);
+
+            // Compute shortest hex distance
+            Hex minSubHex = default;
+            int minDistance = int.MaxValue;
+            foreach (var mainHex in mainHexes)
             {
-                // Add the first parent species organelle
-                AddNewOrganelle(mutatedOrganelles, parentOrganelles[0].Definition);
+                foreach (var islandHex in islandHexes)
+                {
+                    var sub = islandHex - mainHex;
+                    int distance = (Math.Abs(sub.Q) + Math.Abs(sub.Q + sub.R) + Math.Abs(sub.R)) / 2;
+                    if (distance < minDistance)
+                    {
+                        minDistance = distance;
+                        minSubHex = sub;
+
+                        // early exit if distance == 2
+                        if (minDistance == 2)
+                            break;
+                    }
+                }
 
-                // If still empty, copy the first organelle of the parent
-                if (mutatedOrganelles.Count < 1)
-                    mutatedOrganelles.Add((OrganelleTemplate)parentOrganelles[0].Clone());
+                // early exit if distance == 2
```

Comment:
why does distance == 2 mean early exit? And is that the same as minDistance == 2?

Reply:
Updated the comment to make it more clear.
distance 1 == direct neighbor, distance 2 == min distance for island
distance is always >= minDistance, but changed the comment